### PR TITLE
feat(server): gql schema for item CRUD

### PR DESCRIPTION
**Pull request by mimoham24**

### CODE REVIEW

**Review by yk-eukarya — COMMENTED** [review]
Do not forget to configure the resolver in the `gqlgen.yml` file under the `models` section

File: server/schemas/item.graphql [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  model: Model!
+  values: [ItemValue!]!
+  publicVersion: String!
+  #  draftVersion: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
+
+type ItemValue {
+  itemId: ID!
```
Comment (yk-eukarya) [inline server/schemas/item.graphql:13]:
```suggestion
```

**Review by yk-eukarya — APPROVED** [review]
LGTM

File: server/schemas/item.graphql [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+type ItemEdge {
+  cursor: Cursor!
+  node: Item
+}
+
+extend type Query {
+  items(
+    projectId: ID!
```
Comment (rot1024) [inline server/schemas/item.graphql:61]:
```suggestion
    modelId: ID!
```

Remember relation between domain models: `Project *- Model - Schema *- Item`. Also note that this query returns latest versions of each item.

File: server/schemas/item.graphql [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    first: Int
+    last: Int
+    after: Cursor
+    before: Cursor
+  ): ItemConnection!
+  checkItemKeyAvailability(key: String!): KeyAvailability!
```
Comment (rot1024) [inline server/schemas/item.graphql:67]:
```suggestion
```

There should be no need to check the key in the item.

File: server/schemas/item.graphql [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+extend type Mutation {
+  createItem(input: CreateItemInput!): ProjectPayload
+  updateItem(input: UpdateItemInput!): ProjectPayload
+  deleteItem(input: DeleteItemInput!): DeleteProjectPayload
```
Comment (rot1024) [inline server/schemas/item.graphql:73]:
```suggestion
  createItem(input: CreateItemInput!): ItemPayload
  updateItem(input: UpdateItemInput!): ItemPayload
  deleteItem(input: DeleteItemInput!): DeleteItemPayload
```

File: server/schemas/item.graphql [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  modelId: ID!
+  model: Model!
+  values: [ItemValue!]!
+  publicVersion: String!
+  #  draftVersion: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
+
+type ItemValue {
```
Comment (rot1024) [inline server/schemas/item.graphql:12]:
```suggestion
type ItemField {
```

File: server/schemas/item.graphql [hunk 6]
```diff
@@ -0,0 +1,74 @@
+type Item implements Node {
+  id: ID!
+  modelId: ID!
+  model: Model!
+  values: [ItemValue!]!
+  publicVersion: String!
+  #  draftVersion: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
```
Comment (rot1024) [inline server/schemas/item.graphql:10]: Mark "force resolver" for `lastestVersion` and `versions`. `lastestVersion` resolver should returns the latest version if the struct has already `versions`.

```suggestion
  publicVersion: String!
  createdAt: DateTime!
  updatedAt: DateTime!
  lastestVersion: ItemVersion
  versions: [ItemVersion!]!
}

type ItemVersion {
  version: String!
  parent: [String!]!
  ref: [String!]!
  fields: [ItemField!]!
}
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/schemas/item.graphql [hunk 7]
```diff
@@ -0,0 +1,79 @@
+type Item implements Node {
+  id: ID!
+  modelId: ID!
+  model: Model!
+  publicVersion: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  latestVersion: ItemVersion
```
Comment (rot1024) [inline server/schemas/item.graphql:8]:
```suggestion
  createdAt: DateTime!
  updatedAt: DateTime!
  latestVersion: ItemVersion
  publicVersion: ItemVersion
```

**Review by rot1024 — APPROVED** [review]

**Pull request merged**